sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/AppManagedUserAttributeController.java

```diff
@@ -124,22 +124,21 @@ public ResponseEntity<?> addUserAttributesManagedByApp(
             .map(GrantedAuthority::getAuthority)
             .map(String::toUpperCase)
             .collect(Collectors.toList());
-    SugoiUser sugoiUser = new SugoiUser(authentication.getName(), roles);
-    Realm _realm = configService.getRealm(realm);
-    List<String> attributes_allowed =
+    Realm realm = configService.getRealm(realmName);
+    List<String> attributesAllowed =
         Arrays.asList(
-            _realm
+            realm
                 .getProperties()
                 .get(GlobalKeysConfig.APP_MANAGED_ATTRIBUTE_KEYS_LIST)
                 .toUpperCase()
                 .split(","));
     try {
-      if (attributes_allowed.contains(attributeKey.toUpperCase())) {
+      if (attributesAllowed.contains(attributeKey.toUpperCase())) {
 
-        if (permissionService.isWriter(sugoiUser, realm, storage)) {
+        if (permissionService.isWriter(roles, realmName, storage)) {
           ProviderResponse response =
               userService.addAppManagedAttribute(
-                  realm,
+                  realmName,
```

Comment:
Complete this good refactor by factor the service cal and so separate permission time and service time
(idem in delete)